src/core/IIPBase.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * This file is part of ctrl_iip
+ *
+ * Developed for the LSST Data Management System.
+ * This product includes software developed by the LSST Project
+ * (https://www.lsst.org).
+ * See the COPYRIGHT file at the top-level directory of this distribution
+ * for details of code ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "IIPBase.h"
+#include "SimpleLogger.h"
+
+BOOST_LOG_INLINE_GLOBAL_LOGGER_DEFAULT(lg, src::severity_logger_mt< severity_level >);
+#include "IIPMacro.h"
+
+using namespace std; 
+using namespace YAML;
+
+IIPBase::IIPBase(string configfilename, string logfilename) {
+    // Read environment variables
+    this->iip_config_dir = getenv("IIP_CONFIG_DIR");
+    this->iip_log_dir = getenv("IIP_LOG_DIR");
+    this->ctrl_iip_dir = getenv("CTRL_IIP_DIR");
+
+    // Read config file
+    this->l1_config = load_config_file(configfilename);
+    init_log(this->get_log_filepath(), logfilename);
+    LOG_DBG << "Base constructor complete";
+}
+
+string IIPBase::get_log_filepath() { 
+    string path;
+    Node log_node = this->l1_config["LOGGING_DIR"];
+    if (this->iip_log_dir) { 
+        path = this->iip_log_dir;    
+    }
+    else if (log_node) { 
+        path = log_node.as<string>();
+    }
+    else { 
+        path = "/tmp";
+    }
+
+    cout << "Log filepath is " << path << endl;
+    return path;
+}
+
+Node IIPBase::load_config_file(string filename) { 
+    string config_filename;
+    if (filename.empty()) { 
+        config_filename = "L1SystemCfg.yaml"; 
```

Comment:
It's probably better to always explicitly load the file, and never accept an empty name.